SevenSlots/SevenSlots/SevenSlots/Services/DatabaseService.cs

```diff
@@ -0,0 +1,65 @@
+﻿using SevenSlots.Model;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Xamarin.Android.Net;
+
+namespace SevenSlots.Services
+{
+    public class DatabaseService: IDatabaseService
+    {
+        HttpClientHandler httpClientHandler = new HttpClientHandler();
+        HttpClient client;
+        private const string baseUrl = "https://7slotsapi.azurewebsites.net/";
+
+        public async Task<List<User>> getAllUsers()
+        {
+            try
+            {
+                client = new HttpClient(httpClientHandler, true);
+                HttpResponseMessage response = await client.GetAsync(baseUrl + "/User");
+
+                if (!response.IsSuccessStatusCode)
+                {
+                    Console.WriteLine("Fetching the users failed: {0}", response.Content.ReadAsStringAsync());
+                    return null;
+                }
+
+                string content = await response.Content.ReadAsStringAsync();
+                List<User> Users = JsonSerializer.Deserialize<List<User>>(content,
+                    new JsonSerializerOptions() { PropertyNameCaseInsensitive = true});
+
+                return Users;
+            }
+            catch (HttpRequestException e)
+            {
+                Console.WriteLine("Exception Caught! Message :{0} ", e.Message);
+                throw;
```

Comment:
Is this the best solution? To throw the exception?

Reply:
yes

Reply:
Where do you catch it? How do you want to handle this in the future? I don't think we want to let the user crash their own app just because we have an error on the backend